apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Logging/LoggingTest.cs

```diff
@@ -0,0 +1,335 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Diagnostics.Common;
+using Google.Cloud.Diagnostics.Common.Tests;
+using Google.Cloud.Logging.Type;
+using Google.Cloud.Logging.V2;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Xunit;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.IntegrationTests
+{
+    public class LoggingTest
+    {
+        /// <summary>Total time to spend sleeping when looking for log entries.</summary>
+        private readonly TimeSpan _timeout = TimeSpan.FromSeconds(10);
+
+        /// <summary>Time to sleep between checks for log entries.</summary>
+        private readonly TimeSpan _sleepInterval = TimeSpan.FromSeconds(2);
+
+        /// <summary>Project id to run the test on.</summary>
+        private readonly string _projectId;
+
+        /// <summary>Client to use to send RPCs.</summary>
+        private readonly LoggingServiceV2Client _client;
+
+        public LoggingTest()
+        {
+            _projectId = Utils.GetProjectIdFromEnvironment();
+            _client = LoggingServiceV2Client.Create();
+        }
+
+        /// <summary>
+        /// Gets log entries that contain the passed in testId in the log message.  Will poll
+        /// and wait for the entries to appear.
+        /// </summary>
+        /// <param name="startTime">The earliest log entry time that will be looked at.</param>
+        /// <param name="testId">The test id to filter log entries on.</param>
+        /// <param name="minEntries">The minimum number of logs entries that should be waited for.</param>
+        private IEnumerable<LogEntry> GetEntries(DateTime startTime, string testId, int minEntries)
+        {
+            TimeSpan totalSleepTime = TimeSpan.Zero;
+            while (totalSleepTime <= _timeout)
+            {
+                TimeSpan sleepTime = minEntries > 0 ? _sleepInterval : _timeout;
```

Comment:
I'm not clear on why you wait the full timeout amount when minEntries is 0. Is this because 0 is a sentinel value meaning "get all entries for the full timeout"? If so, you may want to call that out in the param docs and/or use a nullable int so null is the sentinel value.